### ISSUE TITLE
Give content creator orientation help 

### ISSUE THREAD

**Issue by bph** [WordPress/gutenberg#50780]
### Description

When I was working with the Details section of the block, there were no indicators if I was still adding content to the details block or already writing outside the details block. 

I expected to leave the Details block when hitting Enter twice similar to the behavior of the List View, but it only added additional empty lines to the details section. 

Only the select mode gives the blue line to determine where the Summary + Details block ends. However, those lines disappear as soon as one clicks to edit the block again. 

To give visual clues to the readers, the content creator only has limited formatting options, like a background for the whole block but not separately for the details section. 

### Step-by-step reproduction instructions

It's hard to come up with exact steps to reproduce, maybe the video helps. 

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/39980/a767054e-dd20-4f01-ac27-729e346de527

### Environment info

WordPress 6.2.1
Gutenberg 15.8
Chrome Version 113.0.5672.92
MacOS Venture 13.3.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
Agree. This was one of the reasons I personally preferred the separate details content block; it was a clearer separation to **me**.
It might help to have the list view open, but that is not always the case.

**Comment by hanneslsm** [comment]
One could argue, that the first icon in the toolbar is the visual indication (parent block selector).
Personally, I am disagreeing  (as stated here https://github.com/WordPress/gutenberg/issues/50021#issuecomment-1527933162), so maybe it makes sense to consider a solution that would not only work for the Details Block.

<img width="330" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/968a1b95-4e61-4e28-842f-dce786daf065">

**Comment by talldan** [comment]
> Personally, I am disagreeing (as stated here https://github.com/WordPress/gutenberg/issues/50021#issuecomment-1527933162), so maybe it makes sense to consider a solution that would not only work for the Details Block.

👍 I also have the same issue in the group block.

**Comment by richtabor** [comment]
The same could be said of the Quote block, or any block with nested capabilities. 

There's not a pattern of changing the visual of a block, other than hover indicators.  I don't think a one-off solution for the details block would resolve the issue. Once you leave the block, it doe close back up (with default behavior). That's one indicator (which maps to the front-end as well).

**Comment by carolinan** [comment]
When pressing enter twice in the new tab panel, the user also stays with in that tab panel.

Group block variations use a dashed line as an indicator, but only when it is empty. This might be "too much":

<img width="704" height="57" alt="Image" src="https://github.com/user-attachments/assets/7408c594-5d72-4d31-8321-bdf17520805a" />
